src/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1409,47 +1422,58 @@ public static String validateCanal(final Route route, final Collection<Unit> uni
    */
   public static String validateCanal(final Territory territory, final Route route, final Collection<Unit> units,
       final PlayerID player, final GameData data) {
+    String failureMessage = null;
     final Set<CanalAttachment> canalAttachments = CanalAttachment.get(territory);
-    if (canalAttachments.isEmpty()) {
-      return null;
-    }
-    final Iterator<CanalAttachment> iter = canalAttachments.iterator();
-    while (iter.hasNext()) {
-      final CanalAttachment attachment = iter.next();
-      if (attachment == null) {
+    for (final CanalAttachment canalAttachment : canalAttachments) {
+      if (!needToCheckCanal(canalAttachment, route, data)) {
```

Comment:
So needToCheckCanal determines if the canal needs to be checked based on if it is actually on the route. If it doesn't need checked then that means its irrelavant (not on the units route) so its skipped. This is due to how canal attachments are attached to territories.